### PR TITLE
[14.0][IMP] sale_product_pack: Set Unit Price to zero for product pack on sale order line when…

### PR DESCRIPTION
**Pull request by cvinh**
… pack component price is detailled
Steps to reproduce : 
- set up a product pack with a price of 1000 EUR tax included : detailed pack type & detailed component price
- set up 2 components products with different tax rates : 
    - 500 EUR 5% tax included
    - 500 EUR 10% tax included
- make a quotation with the product pack and save it

wrong behavior : the total of the sale order is 2000 EUR (each line shows the product price)
right behavior : the total of the sale order should be 1000 EUR (the product pack price should be zero)

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Can you explain the rationale of the change?

**Comment by cvinh** [comment]
> Can you explain the rationale of the change?

Dear @pedrobaeza I updated the PR description.
(I first thought that it was obvious)
Best regards

**Review by pedrobaeza — COMMENTED** [review]
Seems OK. Can you add a test for this use case?

**Comment by cvinh** [comment]
> Seems OK. Can you add a test for this use case?

done

File: sale_product_pack/tests/test_sale_product_pack.py [hunk 1]
```diff
@@ -63,6 +63,8 @@ def test_create_components_price_order_line(self):
             [sequence, sequence, sequence, sequence],
             self.sale_order.order_line.mapped("sequence"),
         )
+        # The price of the product pack line is zero
+        self.assertEqual(pack_line.price_subtotal, 0.0)
```
Comment (pedrobaeza) [inline sale_product_pack/tests/test_sale_product_pack.py]:
```suggestion
        self.assertAlmostEqual(pack_line.price_subtotal, 0.0)
```

**Review by pedrobaeza — APPROVED** [review]

File: sale_product_pack/tests/test_sale_product_pack.py (hunk 1 above)
Comment (cvinh) [reply sale_product_pack/tests/test_sale_product_pack.py]: done

**Comment by cvinh** [comment]
@pedrobaeza I did another PR https://github.com/OCA/product-pack/pull/95 for improving tests with assertAlmostEqual like you suggested

**Comment by pedrobaeza** [comment]
You can close this one then, as the other contains both commits, isn't it?

**Comment by cvinh** [comment]
> You can close this one then, as the other contains both commits, isn't it?

No sorry, I modified the other PR, I did 2 PR to separate cleanly, if you think I can have 2 commits in the same PR I can do it
Please advice

**Comment by pedrobaeza** [comment]
Ok then

**Comment by pedrobaeza** [comment]
I have hit today by a problem with prices (on 13.0) and get to this: #97 , that also solves your problem.

**Comment by cvinh** [comment]
Hello can a psc member apply the 'no stale' label please ?

**Comment by pedrobaeza** [comment]
Closing this one as #97 solves it.

**Pull request closed without merging**